### PR TITLE
Pull in fixes from IpfsDHT

### PR DESCRIPTION
**Pull request by sanderpick**
This uses new (at some point) message request handling. No more request IDs and a bunch of other small changes.

### CODE REVIEW

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 11627e0f30bdf3b522a66afa9e932c297bcce651
  * Commits must be in the following format: **%{scope}: %{description}**



---
This message was auto-generated by https://gitcop.com

File: Makefile [hunk 1]
```diff
@@ -21,6 +21,14 @@ lint:
 build:
 	go build -ldflags "-w" -i -o textile textile.go
 
+cross_build_linux:
```
Comment (carsonfarmer) [inline Makefile]: sweet!

File: service/main.go [hunk 2]
```diff
@@ -134,56 +154,64 @@ func (s *Service) NewEnvelope(mtype pb.Message_Type, msg proto.Message, id *int3
...
 
 // handleError receives an error response
-func (s *Service) handleError(env *pb.Envelope) error {
+func (srv *Service) handleError(env *pb.Envelope) error {
 	if env.Message.Payload == nil && env.Message.Type != pb.Message_PONG {
 		err := fmt.Sprintf("message payload with type %s is nil", env.Message.Type.String())
 		log.Error(err)
 		return errors.New(err)
 	}
+
```
Comment (carsonfarmer) [inline service/main.go]: hah, most of these changes are just formatting and variable name changes! nice :P

File: service/main.go [hunk 3]
```diff
@@ -195,123 +223,120 @@ func (s *Service) handleError(env *pb.Envelope) error {
...
 
-// handleNewMessage handles a p2p net stream
-func (s *Service) handleNewMessage(stream inet.Stream) {
-	defer func() {
-		if err := stream.Close(); err != nil {
-			log.Error(err)
-		}
-	}()
+// handlePing receives a PING message
+func (srv *Service) handlePing(pid peer.ID, env *pb.Envelope) (*pb.Envelope, error) {
```
Comment (carsonfarmer) [inline service/main.go]: cool

File: service/sender.go [hunk 4]
```diff
@@ -14,101 +14,96 @@ import (
...
-	writer     ggio.WriteCloser
-	reader     ggio.ReadCloser
-	mux        sync.Mutex
-	pid        peer.ID
-	service    *Service
-	singleMsg  int
-	invalid    bool
-	requests   map[int32]chan *pb.Envelope
-	requestMux sync.Mutex
+type messageSender struct {
```
Comment (carsonfarmer) [inline service/sender.go]: are these all short/one letter vars because we need this to be small? otherwise, kinda hard to parse as a mere human...

**Comment by carsonfarmer** [comment]
Only really one 'comment' sander, and you can feel free to ignore it, more of a clarification than requirement.

File: service/sender.go (hunk 4 above)
Comment (sanderpick) [reply service/sender.go]: i just went with exactly what's in the `go-libp2p-kad-dht` module as far as variable

**Comment by sanderpick** [comment]
cheers!

File: service/sender.go (hunk 4 above)
Comment (carsonfarmer) [reply service/sender.go]: thought that might be the case. sounds good 👍

File: mobile/mobile.go [hunk 5]
```diff
@@ -122,6 +122,16 @@ func MigrateRepo(config *MigrateConfig) error {
 func NewTextile(config *RunConfig, messenger Messenger) (*Mobile, error) {
 	node, err := core.NewTextile(core.RunConfig{
 		RepoPath: config.RepoPath,
+		LogLevels: map[string]string{
```
Comment (carsonfarmer) [inline mobile/mobile.go]: sweet! just fyi: they all default to error, so you really only need `"tex-service":     "DEBUG"` here...

Comment (sanderpick) [reply mobile/mobile.go]: oh yeah... good catch. i just c/p'd from my cli output
